pandas/tests/io/json/test_pandas.py

```diff
@@ -1731,8 +1732,12 @@ def test_json_pandas_na(self):
         result = DataFrame([[pd.NA]]).to_json()
         assert result == '{"0":{"0":null}}'
 
-    def test_json_pandas_nulls(self, nulls_fixture):
+    def test_json_pandas_nulls(self, nulls_fixture, request):
         # GH 31615
+        if isinstance(nulls_fixture, Decimal):
+            mark = pytest.mark.xfail(reason="not implemented")
```

Comment:
xref #28609 cc @WillAyd

Reply:
If you wanted to fix this here it would just require adding the same condition in the ujson code that we have for checking floats.

https://github.com/pandas-dev/pandas/blob/421fb8d2497910959ebb3c5db6551f01574674ef/pandas/_libs/src/ujson/python/objToJSON.c#L1552

The decimal check is only a few branches below that